### PR TITLE
Implement front-end for confidential queues

### PR DESCRIPTION
**Pull request by jackieo5023**
See #112

### CODE REVIEW

File: src/api/queues.js [hunk 1]
```diff
@@ -139,6 +140,7 @@ router.patch(
       message: data.message !== null ? data.message : undefined,
       messageEnabled:
         data.messageEnabled !== null ? data.messageEnabled : undefined,
+      isConfidential: req.body.isConfidential === true,
```
Comment (nwalters512) [inline src/api/queues.js]: This isn't working right; if I toggle the staff message on a queue, it resets it to being non-confidential. I think once a queue is created as confidential, we don't want to ever allow that to change in the future. We should have a test for this. Let's add a test for the creation endpoint too!

File: src/components/QuestionList.js [hunk 2]
```diff
@@ -158,26 +158,37 @@ class QuestionList extends React.Component {
...
-              didUserAskQuestion={this.props.userId === question.askedById}
-              deleteQuestion={() => this.deleteQuestion(questionId)}
-              cancelQuestion={() => this.cancelQuestion(questionId)}
-              startQuestion={() => this.startQuestion(questionId)}
-              finishedAnswering={() => this.handleFinishedAnswering(questionId)}
-              editQuestion={() => this.handleEditQuestion(questionId)}
-              {...question}
-            />
-          )
+          if (
```
Comment (nwalters512) [inline src/components/QuestionList.js]: We could probably use object destructuring to shorten this a bit:

```js
const { queue, userId } = this.props
if (!queue.isConfidential ...) {}
```

Can also take advantage or short circuiting and get rid of the `queue.isConfidential && ` in the second clause, if my boolean logic is working right 😛

File: src/pages/queue.js [hunk 3]
```diff
@@ -65,7 +75,23 @@ class Queue extends React.Component {
...
+          {this.props.queue.isConfidential && (
+            <span>
+              <FontAwesomeIcon
+                icon={faEyeSlash}
+                fixedWidth
+                className="mr-2"
+                id="confidentialIcon"
+              />
+              <UncontrolledTooltip placement="bottom" target="confidentialIcon">
+                This is a confidential queue! Students won&apos;t be able to see
```
Comment (nwalters512) [inline src/pages/queue.js]: It'd be nice if we could customize this message for students - rephrase it to `You won't be able to see questions from other students`.

Comment (jackieo5023) [reply src/pages/queue.js]: What do you think about just rephrasing the entire message?  So make it `This is a confidential queue! {Students/You} won't be able to see questions from other students.`

Comment (nwalters512) [reply src/pages/queue.js]: I like it!

**Comment by jackieo5023** [comment]
The comments from @nwalters512 should now be implemented

File: src/pages/queue.js [hunk 4]
```diff
@@ -63,9 +73,28 @@ class Queue extends React.Component {
       return <Error statusCode={404} />
     }
     const locationText = this.props.queue.location || 'No location specified'
+    const confidentialMessage = this.props.isUserCourseStaff
```
Comment (nwalters512) [inline src/pages/queue.js]: We'll want a `|| isUserAdmin` here as well so that the message is correct for admins

File: src/components/ConfidentialQueuePanel.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      const numQuestions = this.props.queue.questions.length
+      if (numQuestions === 0) {
+        return null
+      }
+      const cardTotalQuestionsMessage =
+        numQuestions === 1
+          ? 'There is 1 person on the queue'
+          : `There are ${numQuestions} people on the queue`
+      const singularOrPluralPerson = position === 1 ? 'person' : 'people'
+      const cardMessage = questionId
```
Comment (nwalters512) [inline src/components/ConfidentialQueuePanel.js]: This message isn't updating correctly for me. Steps to see the problem:

* Open a browser and log in as a student account
* Add a question: note that the message in that windows says "There is 1 person on the queue with 0 people in front of you." This is correct.
* Open a new browser and log in as another student
* Add a question: note that the message in that windows says "There are 2 people on the queue with 1 person in front of you." This is correct.
* Go back the the original window. The message says "There are 2 people on the queue with 1 person in front of you." This is incorrect, as that user was the first person on the queue and there should be no one in front of them.

File: src/components/ConfidentialQueuePanel.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { ListGroupItem } from 'reactstrap'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSpinner } from '@fortawesome/free-solid-svg-icons'
+
+/* eslint-disable react/prefer-stateless-function */
```
Comment (nwalters512) [inline src/components/ConfidentialQueuePanel.js]: Any reason to not use a function component instead? I don't think we have any state here.

File: src/socket/server.js [hunk 7]
```diff
@@ -29,25 +29,34 @@ const sendInitialState = (queueId, callback) => {
     callback({ questions, activeStaff })
   })
 }
-
+// const blankQuestion = {
```
Comment (nwalters512) [inline src/socket/server.js]: Would this be needed in the future?

Comment (jackieo5023) [reply src/socket/server.js]: Nope, my bad - looks like I didn't delete some old code

File: src/components/ConfidentialQueuePanel.js (hunk 5 above)
Comment (jackieo5023) [reply src/components/ConfidentialQueuePanel.js]: Found out I was doing a foreach over the questions and was never quitting when the user question was found... this should now be fixed

File: src/components/ConfidentialQueuePanel.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
```
Comment (nwalters512) [inline src/components/ConfidentialQueuePanel.js]: We can get rid of the eslint-disable directive here
